lib/esereader.h

```diff
@@ -36,12 +36,15 @@ class ESEReader {
   int32_t         readSize () { return m_readSize; }
   virtual int32_t streamSize () { return 0; }
   int32_t         streamPosition () { return m_streamPosition; }
+  uint32_t        bufferReadLength () { return m_bufferReadLength; }
```

Comment:
This class has sort of become a hybrid. It would be useful to have a pure reader class just concerned with `{len(), pos(), data[]}`, and then buffer atop.  Go has a lovely design https://pkg.go.dev/bufio but I really think we could simplify this a lot just reusing the C++ standard library.

Reply:
I tried to make it as pure as possible. Next PR